releases/release-5.4.0.md

```diff
@@ -0,0 +1,348 @@
+
+- **在 TiDB Data Migration (DM) 中 优化 relay log 的使用方式**
+
+    - 恢复 `source` 配置中 `enable-relay` 开关
+    - 增加 `start-relay` 或 `stop-relay` 命令中动态开启或关闭 relay log 的功能
+    - relay log 的开启状态与 `source` 绑定，source 迁移到任意 DM-worker 均保持原有开启或关闭状态
+    - relay log 的存放路径移至 DM-worker 配置文件
+
+- **在 DM 中优化[排序规则](/character-set-and-collation.md)的处理方式**
+
+    增加 collation_compatible 开关，支持 strict 和 loose（默认）两种模式：
+
+    - 如果对排序规则要求不严格，允许上下游查询结果排序规则不一致，使用默认的 loose 模式可以避免报错。
+    - 如果对排序规则要求严格，业务要求排序规则必须一致，则应当使用 strict 模式。但如果下游不支持上游缺省的 collation，同步可能会报错。
+
+- **在 DM 中优化 `transfer source`，支持平滑执行同步任务**
+
+    当 DM-worker 所在各节点负载不均衡时，`transfer source` 命令可用于手动将某 `source` 配置迁移到其他节点。优化后的 `transfer source` 简化了用户操作步骤，不再要求先暂停所有关联 task 而是直接执行平滑迁移，DM 将在内部完成所需操作。
+
+- **DM OpenAPI 特性 GA**
+
+    DM 支持通过 API 的方式进行日常管理，包括增加数据源、管理任务等。本次更新 OpenAPI 从实验特性转为正式特性。
+
+### 问题诊断效率
+
+- **Top SQL（实验特性）**
+
+    新推出实验性特性 Top SQL（默认关闭），帮助用户轻松找到节点中消耗负载较大的查询。
+
+    [用户文档](/dashboard/top-sql.md)
+
+### TiDB 数据共享订阅
+
+- **优化 TiCDC 对集群的影响**
+
+    大幅降低了 TiCDC 启用后，对 TiDB 集群的性能影响。在实验室环境中，TiCDC 对 TiDB 的性能影响可以降低到 5% 以下。
+
+### 部署及运维
+
+- **持续性能分析（实验特性）**
+
+    - 支持更多组件：除了 TiDB、PD 和 TiKV 外，v5.4.0 版本中还支持查看 TiFlash CPU Profiling
+    - 支持更方便的查看形式：支持以火焰图形式查看 CPU Profiling 和 Goroutine 结果。
+    - 支持更多部署环境：支持在 TiDB Operator 部署环境下启用持续性能分析功能。
+
+    该功能默认关闭，需进入 TiDB Dashboard 持续性能分析页面开启，开启方法见[用户文档](/dashboard/continuous-profiling.md)。
+
+    持续性能分析仅支持由 v1.9.0 及以上版本 TiUP 或 v1.3.0 及以上版本 TiDB Operator 升级或安装的集群。
+
+## 提升改进
+
++ TiDB
+
+    - 新增系统变量 `tidb_enable_paging`，开启该功能可显著降低使用 `IndexLookUp` 和 `Limit` 并且 `Limit` 数据较小且无法下推到 `IndexScan` 上的读请求的延迟 [#30578](https://github.com/pingcap/tidb/issues/30578)
+    - 支持 `ADMIN {SESSION | INSTANCE | GLOBAL} PLAN_CACHE` 语法，用于清空缓存的查询计划 [#30370](https://github.com/pingcap/tidb/pull/30370)
+
++ TiKV
+
+    - Coprocessor 支持分页 API 进行流式处理 [#11448](https://github.com/tikv/tikv/issues/11448)
+    - 支持 `read-through-lock`，使读操作不需要等待清理 secondary lock [#11402](https://github.com/tikv/tikv/issues/11402)
+    - 增加了磁盘保护机制，尽量避免磁盘空间耗尽导致 panic [#10537](https://github.com/tikv/tikv/issues/10537)
+    - 日志支持存档和轮替 [#11651](https://github.com/tikv/tikv/issues/11651)
+    - 减少 Raft 客户端的系统调用并提高 CPU 效率 [#11309](https://github.com/tikv/tikv/issues/11309)
+    - Coprocessor 支持下推 substring 到 TiKV [#11495](https://github.com/tikv/tikv/issues/11495)
+    - 通过跳过读锁提高在 RC 隔离级别中扫描的性能 [#11485](https://github.com/tikv/tikv/issues/11485)
+    - 减少备份使用的默认线程池大小，并在压力大时限制其使用 [#11000](https://github.com/tikv/tikv/issues/11000)
+    - 支持动态调整 Apply 和 Store 线程池大小 [#11159](https://github.com/tikv/tikv/issues/11159)
+    - 支持配置 `snap-generator` 线程池大小 [#11247](https://github.com/tikv/tikv/issues/11247)
+    - 优化在文件数较多且读写频繁的场景下 RocksDB 的全局锁争用问题 [rocksdb#250](https://github.com/tikv/rocksdb/pull/250)
+
++ PD
+
++ TiFlash
+
++ Tools
+
+    + Backup & Restore (BR)
+
+    + TiCDC
```

Comment:
## Bug 修复

* 修复当 `min.insync.replicas` 小于 `replication-factor` 时不能同步的问题 [#3994](https://github.com/pingcap/tiflow/issues/3994)
* 修复 cached region 监控指标为负数的问题 [#4300](https://github.com/pingcap/tiflow/issues/4300)
* 修复 mq sink write row 没有监控数据的问题 [#3431](https://github.com/pingcap/tiflow/issues/3431)
* 修复 sql mode 兼容性问题 [#3810](https://github.com/pingcap/tiflow/issues/3810)
* 修复在移除同步任务后潜在的 panic 问题 [#3128](https://github.com/pingcap/tiflow/issues/3128)
* 修复输出默认列值时的 panic 问题和数据不一致问题 [#3929](https://github.com/pingcap/tiflow/issues/3929)
* 修复不支持同步默认值的问题 [#3793](https://github.com/pingcap/tiflow/issues/3793)
* 修复潜在的同步流控死锁问题 [#4055](https://github.com/pingcap/tiflow/issues/4055)
* 修复在磁盘写满时无日志输出的问题 [#3362](https://github.com/pingcap/tiflow/issues/3362)
* 修复 DDL 特殊注释导致的同步停止的问题 [#3755](https://github.com/pingcap/tiflow/issues/3755)
* 修复在某些 RHEL 发行版上因时区问题导致服务无法启动的问题 [#3584](https://github.com/pingcap/tiflow/issues/3584)
* 修复因 checkpoint 不准确导致的潜在的数据丢失问题 [#3545](https://github.com/pingcap/tiflow/issues/3545)
* 修复在容器环境中 OOM 的问题 [#1798](https://github.com/pingcap/tiflow/issues/1798)
* 修复 config.Metadata.Timeout 没有正确配置而导致的同步停止问题 [#3352](https://github.com/pingcap/tiflow/issues/3352)

## 提升改进

* 减少 "EventFeed retry rate limited" 日志 [#4006](https://github.com/pingcap/tiflow/issues/4006)
* 降低在同步大量表时的同步延时 [#3900](https://github.com/pingcap/tiflow/issues/3900)
* 减低 TiKV 节点宕机后 kvclient 恢复的时间 [#3191](https://github.com/pingcap/tiflow/issues/3191)

Reply:
附上英语版本

## Bug 修复

* Fix the problem that TiCDC cannot send messages when `min.insync.replicas` is less than `replication-factor` [#3994](https://github.com/pingcap/tiflow/issues/3994)
* Fix kv client cached region metric could be negative. [#4300](https://github.com/pingcap/tiflow/issues/4300)
* Fix mq sink write row count metrics. [#3431](https://github.com/pingcap/tiflow/issues/3431)
* Fix issues about sql mode compatibility for mysql sink. [#3810](https://github.com/pingcap/tiflow/issues/3810)
* Fix the potential panic issue that occurs when changefeed info is removed from etcd. [#3128](https://github.com/pingcap/tiflow/issues/3128)
* Fix column default value panic and data inconsistency. [#3929](https://github.com/pingcap/tiflow/issues/3929)
* Fix mounter default date value not support [#3793](https://github.com/pingcap/tiflow/issues/3793)
* Fix a bug that can cause changefeed stuck due to a deadlock occurs. [#4055](https://github.com/pingcap/tiflow/issues/4055)
* Fix TiCDC doesn't report any error when log disk is full. [#3362](https://github.com/pingcap/tiflow/issues/3362)
* Fix syntax error if DDL has a special comment. [#3755](https://github.com/pingcap/tiflow/issues/3755)
* Fix timezone related error that cause cdc server can't run in some RHEL release version (6.8, 6.9 etc). [#3584](https://github.com/pingcap/tiflow/issues/3584)
* Fix data lose due to inaccurate checkpoints. [#3545](https://github.com/pingcap/tiflow/issues/3545)
* Fix OOM in container environment [#1798](https://github.com/pingcap/tiflow/issues/1798)
* Set config.Metadata.Timeout correctly to prevent stuck data synchronization. [#3352](https://github.com/pingcap/tiflow/issues/3352)

## 提升改进

* Reduce "EventFeed retry rate limited" logs [#4006](https://github.com/pingcap/tiflow/issues/4006)
* Reduce checkpoint lag when capturing many tables. [#3900](https://github.com/pingcap/tiflow/issues/3900)
* Reduce kvclient recover time when TiKV is down. [#3191](https://github.com/pingcap/tiflow/issues/3191)